source/elements/oneDAL/source/common_interface/header_files_and_namespaces.rst

```diff
@@ -1,3 +1,61 @@
-===========================
-Header files and namespaces
-===========================
+============
+Header files
+============
+
+oneDAL public identifiers are represented in the following header files:
+
+.. list-table::
+   :widths:  10 50
+   :header-rows: 1
+
+   * - header file
+     - description
+   * - ``onedal/onedal.hpp```
+     - Main header file of oneDAL library.
+   * - ``onedal/algo/%ALGORITHM%/%ALGORITM%.hpp``
+     - An algorithm header file for a particular algorithm.
+       The string %ALGORITHM% should be substituted with the name of the algorithm, for example, ``kmeans`` or ``knn``.
+
+   * - ``onedal/algo/misc/%FUNCTIONALITY%/%FUNCTIONALITY%.hpp``
+     -  A header file for miscellaneous data types and functionality intended to be used by oneDAL algorithms and applications for algorithm customization and optimization on various stages
+        of the analytical pipeline.
+        The string %FUNCTIONALITY% should be substituted with the functionality name, for example, ``mt19937`` or ``cross_enthropy_loss``.
+
+   * - ``onedal/util/%UTILITY%.hpp```
+     - A header file for auxiliary functionality intended to be used for design of classes and implementation of methods such as memory allocators or type traits. 
```

Comment:
```suggestion
     - A header file for auxiliary functionality that is intended to be used to design classes and implement such methods as memory allocators and type traits. 
```

Reply:
The expected meaning is: things like memory allocators and type traits form auxiliary functionality which is used to design  ...

Reply:
would this work?

```suggestion
     - A header file for auxiliary functionality intended to be used to design classes that implement such methods as memory allocators and type traits. 
```